CONTRIBUTING.md

```diff
@@ -13,53 +13,114 @@ At the same time, I had a day job where I was a junior software developer at a r
 - **Making friends in the larger community:** The creator of request, [@mikeal](https://twitter.com/mikeal), is still a friend to this day.
 
-If any of this sounds interesting, I hope you consider getting involved with Astro. Come say hi in the [**#new-contributors**](https://astro.build/chat) channel on Discord, anytime. We're always around and value contributions of any shape/size.
+If any of this sounds interesting, I hope you consider getting involved with Astro. Come say hi in the [**#new-contributors**][discord] channel on Discord anytime. We're always around and value contributions of any shape/size.
 
 # Contributor Manual
 
 ## Prerequisite
 
-```shell
-node: "^12.20.0 || ^14.13.1 || >=16.0.0"
-yarn: "^1.22.10"
-# otherwise, your build will fail
-```
+You’ll need the following tools set up on your system to contribute to this codebase:
 
-## Setting up your local repo
+- [Node 14.x or 16.x][install-node]
+- [Yarn 1.x][install-yarn] (_Note: we do NOT use Yarn 2!_)
+- [Git 2.x][install-git]
 
-Astro uses yarn workspaces, so you should **always run `yarn install` from the top-level project directory.** running `yarn install` in the top-level project root will install dependencies for `astro`, `www`, `docs`, and every package in the repo.
+## Setting up Astro locally
 
-```shell
-git clone && cd ...
-yarn install
-yarn build:all
+All work on Astro begins by downloading a copy locally. Start by [**forking**][how-to-fork] this repository (if you’re new to this, [GitHub has a great detailed guide][how-to-fork]). You should end up with an `astro/` code folder on your local machine.
+
+Switch to this folder by opening your **Terminal application** of choice and running:
+
+```
+cd /[path to my code folder]/astro
 ```
 
+Next, you’ll need to install dependencies and set up the project. You can do that by running the following commands from within the `astro/` folder:
+
+```
+yarn
+yarn build
+```
+
+This will download everything you need and build Astro locally for use. You’re ready to go!
+
 ## Development
 
-```shell
-# starts a file-watching, live-reloading dev script for active development
+While working within packages (such as `packages/astro/*`), you may notice you have to run `yarn build` to see your changes. This process can really slow you down if you are trying many changes at once. Try running the following instead:
+
+```
 yarn dev
-# build the entire project, one time.
+```
+
+This will start a build watcher, which will rebuild files as you work. Use this to speed up your workflow, and see your changes more rapidly.
+
+Of course, you can still build all files within a package, as before. You may need to do so from time to time:
+
+```
 yarn build
 ```
 
 ## Running tests
 
-```shell
-# run this in the top-level project root to run all tests
+You’ll need to run existing tests to make sure your work didn’t cause any regressions. To create a fresh build and run all tests, run the following from the project root:
+
+```
+yarn
```

Comment:
nit: `yarn` only needs to be run once, and `yarn build` isn't needed when `yarn dev` is already running. Is there a way to communicate this so that users aren't running 3 commands every time they test?

Reply:
> `yarn build` isn't needed when `yarn dev` is already running. 

There’s actually a lot of cases where this isn’t true. 2 main scenarios are if files were created, deleted, or renamed, `yarn build` needs to be run. Also, there’s a bug in our dev scripts where some post-build actions are skipped, resulting in errors (e.g. the `postbuild` script in `astro-markdown`). For beginners, I think this is confusing, so I just recommend running a fresh (correct) build before running tests.

Reply:
Totally agreed if its been a while, but my fear is that the docs seem to suggest this flow:

- make a change
- run `yarn`
- run `yarn build`
- run `yarn test`
- make another change
- run `yarn`
- run `yarn build`
- etc. etc.

Can you communicate what you want to communicate, in a way that just makes it really clear that `yarn` and `yarn build` aren't needed every time?

Reply:
Perhaps I’ll just add a section outlining all the scripts separately and reference back to them.